proto/regen/ecocredit/basket/v1/query.proto

```diff
@@ -0,0 +1,97 @@
+syntax = "proto3";
+
+package regen.ecocredit.basket.v1;
+
+option go_package = "github.com/regen-network/regen-ledger/x/ecocredit/basket";
+
+import "google/api/annotations.proto";
+import "regen/ecocredit/basket/v1/state.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+
+// Msg is the regen.ecocredit.basket.v1beta1 Query service.
+service Query {
+
+    // Basket queries one basket by denom.
+    rpc Basket(QueryBasketRequest) returns (QueryBasketResponse) {
+        option (google.api.http).get =
+            "/regen/ecocredit/basket/v1/baskets/{basket_denom}";
+    }
+
+    // Baskets lists all baskets in the ecocredit module.
+    rpc Baskets(QueryBasketsRequest) returns (QueryBasketsResponse) {
+        option (google.api.http).get =
+            "/regen/ecocredit/basket/v1/baskets";
+    }
+
+    // BasketBalances lists the balance of each credit batch in the basket.
+    rpc BasketBalances(QueryBasketBalancesRequest) returns (QueryBasketBalancesResponse) {
+        option (google.api.http).get =
+            "/regen/ecocredit/basket/v1/baskets/{basket_denom}/balances";
+    }
+
+    // BasketBalanceByDenom queries the balance of a specific credit batch in the basket.
+    rpc BasketBalanceByDenom(QueryBasketBalanceByDenomRequest) returns (QueryBasketBalanceByDenomResponse) {
+        option (google.api.http).get =
+            "/regen/ecocredit/basket/v1/baskets/{basket_denom}/balances/{batch_denom}";
```

Comment:
```suggestion
            "/regen/ecocredit/basket/v1/baskets/{basket_denom}/balance/{batch_denom}";
```

Reply:
i would prefer to keep it this way to align with RESTful resource conventions. we already have the `balances` collection in the previous query so would prefer to keep it the way it is.

ref: https://restfulapi.net/resource-naming/ - 1.1. Singleton and Collection Resources

Reply:
OK, thanks for the link